src/tiledimage.js

```diff
@@ -1784,22 +1810,21 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
             levelVisibility
         );
 
-        if (!tile.loaded) {
-            if (tile.context2D) {
-                this._setTileLoaded(tile);
-            } else {
-                var imageRecord = this._tileCache.getImageRecord(tile.cacheKey);
-                if (imageRecord) {
-                    this._setTileLoaded(tile, imageRecord.getData());
-                }
-            }
+        // Try-find will populate tile with data if equal tile exists in system
+        if (!tile.loaded && !tile.loading && this._tryFindTileCacheRecord(tile)) {
+            loadingCoverage = true;
         }
 
         if ( tile.loading ) {
             // the tile is already in the download queue
             this._tilesLoading++;
         } else if (!loadingCoverage) {
-            best = this._compareTiles( best, tile, this.maxTilesPerFrame );
+            // add tile to best tiles to load only when not loaded already
+            best = this._compareTiles( best, tile, this._currentMaxTilesPerFrame );
+            // TODO: test 'Viewer headers can be updated' fail if we start decreasing the number since not enough tiles get invoked
```

Comment:
This is an attempt to fall back after some time (after tiledimage reset or when created) to the default value, but start from a higher number. And tests fail. there is a test that expects 4+ tiles load trigger, and gets 2 with the async api. So it times out. This is the penalization I was talking about. So maybe we can actually leave the constant bigger, since this test shows that it indeed loads less tiles.

Reply:
Odd. Loading more tiles at the beginning and then going down to the value set by the user seems like a good strategy. Maybe it needs to be handled differently somehow? Or maybe the test needs to be modified? Just because the test loads fewer tiles doesn't mean it actually is better to leave it big for all user scenarios.

Reply:
I will change the test then..